tools/runner

```diff
@@ -78,10 +78,11 @@ if args.version:
 test = os.path.abspath(os.path.join(dirs['tests'], args.test))
 
 req_test_params = [
-    "name", "tags", "description", "should_fail", "files", "incdirs",
-    "top_module", "timeout", "type"
+    "name", "tags", "description", "files", "incdirs", "top_module", "timeout",
+    "type"
 ]
 
+optional_test_params = ["should_fail", "should_fail_because"]
```

Comment:
I think that `should_fail` and `should_fail_because` should be part of `req_test_params` and made optional by providing default values in `else` clause, ie. `0` for `should_fail` if `should_fail_because` was not provided and an empty string for `should_fail_because`.